### PR TITLE
libpfring: add new package

### PR DESCRIPTION
**Pull request by hbl0307106015**
PF_RING is a high speed packet capture library that turns a commodity PC into an efficient and cheap
network measurement box suitable for both packet and active traffic analysis and manipulation.
Moreover, PF_RING opens totally new markets as it enables the creation of efficient application such as
traffic balancers or packet filters in a matter of lines of codes.

Signed-off-by: BangLang Huang <banglang.huang@foxmail.com>

official: http://www.ntop.org/products/packet-capture/pf_ring/
github: https://github.com/ntop/pf_ring

Maintainer: me
Compile tested: x86_64, x86_64 relevant machine, LEDE latest
Run tested: x86_64, x86_64 relevant machine, LEDE latest, tests done

### CODE REVIEW

**Comment by hbl0307106015** [comment]
Pull request V2:

fix DEPENDS for x86_64
fix no description
fix duplicated Build/Configure
add URL details in commit message

**Comment by hbl0307106015** [comment]
Pull request V3:

bump to latest version
the 2 patch file has been merged by upstream 

https://github.com/ntop/PF_RING/commit/549a80dd3dbbaf3422ff5ce36920fa9c2f2149e9

@hnyman

**Comment by feckert** [comment]
This is only the pf_ring userland lib or?
Is the kernel module explained in the documentation upstream or will this also installed by this target?

**Comment by hbl0307106015** [comment]
Yes,  this is only the userland lib currently, I will update my PR v4 with the kernel module soon, thanks for your hint. @feckert

**Comment by hbl0307106015** [comment]
Pull request V4:

bump to latest version
add missing kernel package for pf_ring

@hnyman

File: kernel/pfring/Makefile [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+PKG_NAME:=pf-ring
+PKG_VERSION:=git-ce37e8d
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://github.com/ntop/pf_ring.git
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
+PKG_SOURCE_VERSION:=ce37e8d2bef24dfc8f9cf28b7b1387276928cbd5
```
Comment (hauke) [inline kernel/pfring/Makefile]: Please add a PKG_MIRROR_HASH

File: kernel/pfring/Makefile [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+include $(INCLUDE_DIR)/kernel.mk
+
+PKG_NAME:=pf-ring
+PKG_VERSION:=git-ce37e8d
```
Comment (hauke) [inline kernel/pfring/Makefile]: I would suggest to use the version 6.6.0  now

Comment (hbl0307106015) [inline kernel/pfring/Makefile]: OK

**Comment by hbl0307106015** [comment]
Hi all, @hauke @champtar @hnyman  My Pull Request has been updated, the version bump to latest stable version 6.6.0 and use PKG_HASH suggested by @hauke , please check it again, thanks.

**Comment by hbl0307106015** [comment]
Pull request V5:

bump to latest stable version 6.6.0
add missing PKG_HASH variable

**Comment by champtar** [comment]
This is not building on ar71xx (see Travis)

**Comment by hbl0307106015** [comment]
@champtar  This package is only for target x86_64 and I wrote DEPENDS:=@Target_x86_64. why it can build on ar71xx?

**Comment by hnyman** [comment]
Travis script uses always ar71xx as the test platform, so it will always fail packages that can't be built on ar71xx. (like this)

**Comment by hbl0307106015** [comment]
@hnyman OK, thank you for your reply. Please feel free to let me know if there's any mistake on my commits @diizzyy . :-)

File: kernel/pfring/Makefile [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+CONFIGURE_PATH:=kernel
+MAKE_PATH:=kernel
+
+define KernelPackage/pf-ring
+  SUBMENU:=Network Support
+  TITLE:=PF_RING Kernel driver
+  FILES:=$(PKG_BUILD_DIR)/kernel/pf_ring.ko
+  AUTOLOAD:=$(call AutoLoad,90,pf_ring,1)
+  DEPENDS:=+libpf-ring
```
Comment (champtar) [inline kernel/pfring/Makefile]: This seems inverted to me, isn't libpf-ring depending on kernel pf-ring ?

File: kernel/pfring/Makefile [hunk 4]
```diff
@@ -0,0 +1,66 @@
+#
+# Copyright (C) 2017 OpenWrt.org
```
Comment (diizzyy) [inline kernel/pfring/Makefile]: Don't assign it to OpenWrt unless you're affiliated, use your own name or omit it.

File: kernel/pfring/Makefile [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+#
+
+include $(TOPDIR)/rules.mk
+include $(INCLUDE_DIR)/kernel.mk
+
+PKG_NAME:=pf-ring
+PKG_VERSION:=6.6.0
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_VERSION).tar.gz
```
Comment (diizzyy) [inline kernel/pfring/Makefile]: No, this is a showstopper. Use a git checkout instead and tar.xz as compression.

File: libs/libpfring/Makefile [hunk 6]
```diff
@@ -0,0 +1,73 @@
+#
+# Copyright (C) 2017 OpenWrt.org
```
Comment (diizzyy) [inline libs/libpfring/Makefile]: Same as above

File: libs/libpfring/Makefile [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=libpf-ring
+PKG_VERSION:=6.6.0
+PKG_RELEASE:=1
+PKG_MAINTAINER:=Banglang Huang <banglang.huang@foxmail.com>
+
+PKG_SOURCE:=$(PKG_VERSION).tar.gz
```
Comment (diizzyy) [inline libs/libpfring/Makefile]: Same as above

**Comment by diizzyy** [comment]
tarball filename(s) needs to be fixed before merging.

**Comment by hbl0307106015** [comment]
@diizzyy  @hnyman I updated my commit, please check it again. :-)

**Comment by diizzyy** [comment]
@hbl0307106015 
License tag is wrong but other than that it looks fine to me.
https://spdx.org/licenses/LGPL-2.1.html

**Comment by hbl0307106015** [comment]
@diizzyy Got it, I updated kernel module for GPL-2.0 and library for LGPL-2.1.

**Comment by hbl0307106015** [comment]
https://github.com/ntop/PF_RING/pull/286

**Comment by hbl0307106015** [comment]
ChangeLog:

1. bump to 7.0.0 stable version

**Comment by hbl0307106015** [comment]
Pull request V7:

1. bump to 7.0.0 latest commit
2. fix compilation error on non-x86 target.

**Comment by stintel** [comment]
@hbl0307106015 would you mind bumping this to the last release? I want to try this with suricata. If it works without issues I'll finally accept this PR.

**Comment by stintel** [comment]
Oh and the kmod package is missing this:
```define Build/InstallDev
       $(INSTALL_DIR) $(1)/usr/include/linux
       $(CP) $(PKG_INSTALL_DIR)/usr/include/linux/*.h $(1)/usr/include/linux

endef```

**Comment by stintel** [comment]
And I would prefer that PKG_NAME and the directory containing the Makefile have the same name. Since the library is name libpfring.so, I would use libpfring as PKG_NAME. Does this make sense?

**Comment by stintel** [comment]
One more issue I have: the SONAME of libpfring.so is libpfring.so.1, but the file is just libpfring.so. Suricata binary wants libpfring.so.1, which does not exist. Not sure if this is a bug in libpfring or in suricata..

**Comment by hbl0307106015** [comment]
@stintel  OK, I will bump it to latest version.

**Comment by diizzyy** [comment]
@hbl0307106015 
Please switch to codeload

**Comment by hbl0307106015** [comment]
@diizzyy  sorry, what's codeload ??

@stintel  It works fine now , please check it again, thanks!
1. add libpfring.so.1
2. bump to 7.2.0 stable version
3. rename PKG_NAME

**Comment by diizzyy** [comment]
@hbl0307106015 
https://github.com/openwrt/packages/pull/6691/files
That should give you an idea, it packages the source code tree for you

**Comment by hbl0307106015** [comment]
@diizzyy  OK, switched to codeload, thanks!

**Comment by hbl0307106015** [comment]
Hi @stintel, Is there any issue with my commit ?

**Comment by stintel** [comment]
Currently unable to test it due to #6751 breaking my builds.

**Comment by stintel** [comment]
Renamed kmod-pfring back to kmod-pf-ring, and merged this in bc01c8eae6ef0d87b1a0e6ac676605cacbd52014. Thanks for your contribution!

**Pull request closed without merging**